airbyte-ci/connectors/live-tests/src/live_tests/commons/models.py

```diff
@@ -349,10 +350,12 @@ def get_status_messages_per_stream(self, stream: str) -> Dict[str, List[AirbyteS
         return statuses
 
     def get_message_count_per_type(self) -> dict[AirbyteMessageType, int]:
-        message_count: dict[AirbyteMessageType, int] = defaultdict(int)
-        for message in self.airbyte_messages:
-            message_count[message.type] += 1
-        return message_count
+        if self._message_count_per_type is None:
```

Comment:
What do you think about using a [`@cache` decorators](https://docs.python.org/3/library/functools.html#functools.cache) on some getters declared in `ExecutionResult` (basically the getters with small memory footprint) , it would avoid requiring the declaration of private nullable attributes.

Reply:
I really like this idea. I actually tried it here but got an error due to an invalid cache key. Will look a little more closely at whether this could be viable with a slight reorganization of the code.

Reply:
@alafanechere I've modified this to use `@cache`, by adding a `__cache__` method to `ExecutionResult`. LMK what you think.